src/test/resources/org/rust/lang/core/parser/fixtures/well-formed/way_too_many_parens.rs

```diff
@@ -0,0 +1,19 @@
+static i: () =
+(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
+()
+)))))))))))))))))))))))))))))))))))))))))))))))))))))))))))
+;
+
+static j: i32 =
+(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
+1
+)))))))))))))))))))))))))))))))))))))))))))))))))))))))))))
+;
+
+static k: (i32,) =
+(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((
+1,
+)))))))))))))))))))))))))))))))))))))))))))))))))))))))))))
+;
+
+fn main() {}
```

Comment:
Nice idea! Especially because it requires to tests the same thing in the types part of grammar. We don't have this problem there though, because we don't distinguish parenthesized and tuple types. 

fixed